pkg/api/vm/handler.go

```diff
@@ -320,6 +325,10 @@ func (h *vmActionHandler) migrate(ctx context.Context, namespace, vmName string,
 	if !canMigrate(vmi) {
 		return errors.New("The VM is already in migrating state")
 	}
+	if isSpecificNodeFromVMINodeSelector(vmi) {
```

Comment:
check node selector in canMigrate.

---
for the vmi has node selector case,
UI will call findMigratableNodes and report an error when users click the `Migrate` button.
![image](https://github.com/harvester/harvester/assets/15064560/15991159-5470-41e9-a07d-ece3a5375360)
if we check node selector in `canMigrate`, the `Migrate` button will be hidden